File: app/src/main/java/torille/fi/lurkforreddit/subreddits/SubredditsPresenterModule.java
package torille.fi.lurkforreddit.subreddits;

import dagger.Module;
import dagger.Provides;
import torille.fi.lurkforreddit.data.RedditRepository;

/**
 * Created by eva on 24.4.2017.
 */
@Module
public class SubredditsPresenterModule {

    public SubredditsPresenterModule() {
    }

    @Provides
    public SubredditsContract.Presenter<SubredditsContract.View> provideSubredditsPresenter(RedditRepository redditRepository) {
        return new SubredditsPresenter(redditRepository);
    }

}

File: app/src/main/java/torille/fi/lurkforreddit/subreddits/SubredditsPresenterModule.kt
package torille.fi.lurkforreddit.subreddits

import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector
import torille.fi.lurkforreddit.di.scope.FragmentScoped


/**
 * This is a Dagger module. We use this to pass in the View dependency to the
 * [SubredditsPresenter].
 */
@Module
abstract class SubredditsPresenterModule {

    @FragmentScoped
    @ContributesAndroidInjector
    abstract fun subredditsFragment(): SubredditsFragment

    @FragmentScoped
    @Binds
    abstract fun subredditsPresenter(presenter: SubredditsPresenter): SubredditsContract.Presenter

}
